runtime/src/prioritization_fee_cache.rs

```diff
@@ -0,0 +1,788 @@
+    total_update_elapsed_us: AtomicU64,
+
+    // Accumulated time spent on acquiring cache write lock.
+    total_cache_lock_elapsed_us: AtomicU64,
+
+    // Accumulated time spent on acquiring each block entry's lock..
+    total_entry_lock_elapsed_us: AtomicU64,
+
+    // Accumulated time spent on updating block priotization fees.
+    total_entry_update_elapsed_us: AtomicU64,
+
+    // Accumulated time spent on finalizing block priotization fees.
+    total_block_finalize_elapsed_us: AtomicU64,
+}
+
+impl PrioritizationFeeCacheMetrics {
+    fn accumulate_successful_transaction_update_count(&self, val: u64) {
+        self.successful_transaction_update_count
+            .fetch_add(val, Ordering::Relaxed);
+    }
+
+    fn accumulate_total_update_elapsed_us(&self, val: u64) {
+        self.total_update_elapsed_us
+            .fetch_add(val, Ordering::Relaxed);
+    }
+
+    fn accumulate_total_cache_lock_elapsed_us(&self, val: u64) {
+        self.total_cache_lock_elapsed_us
+            .fetch_add(val, Ordering::Relaxed);
+    }
+
+    fn accumulate_total_entry_lock_elapsed_us(&self, val: u64) {
+        self.total_entry_lock_elapsed_us
+            .fetch_add(val, Ordering::Relaxed);
+    }
+
+    fn accumulate_total_entry_update_elapsed_us(&self, val: u64) {
+        self.total_entry_update_elapsed_us
+            .fetch_add(val, Ordering::Relaxed);
+    }
+
+    fn accumulate_total_block_finalize_elapsed_us(&self, val: u64) {
+        self.total_block_finalize_elapsed_us
+            .fetch_add(val, Ordering::Relaxed);
+    }
+
+    fn report(&self, slot: Slot) {
+        datapoint_info!(
+            "block_prioritization_fee_counters",
+            ("slot", slot as i64, i64),
+            (
+                "successful_transaction_update_count",
+                self.successful_transaction_update_count
+                    .swap(0, Ordering::Relaxed) as i64,
+                i64
+            ),
+            (
+                "total_update_elapsed_us",
+                self.total_update_elapsed_us.swap(0, Ordering::Relaxed) as i64,
+                i64
+            ),
+            (
+                "total_cache_lock_elapsed_us",
+                self.total_cache_lock_elapsed_us.swap(0, Ordering::Relaxed) as i64,
+                i64
+            ),
+            (
+                "total_entry_lock_elapsed_us",
+                self.total_entry_lock_elapsed_us.swap(0, Ordering::Relaxed) as i64,
+                i64
+            ),
+            (
+                "total_entry_update_elapsed_us",
+                self.total_entry_update_elapsed_us
+                    .swap(0, Ordering::Relaxed) as i64,
+                i64
+            ),
+            (
+                "total_block_finalize_elapsed_us",
+                self.total_block_finalize_elapsed_us
+                    .swap(0, Ordering::Relaxed) as i64,
+                i64
+            ),
+        );
+    }
+}
+
+enum FinalizingServiceUpdate {
```

Comment:
Seems like this needs a new name, now that it includes non-finalizing variants

Reply:
good call, changed to `CacheServiceUpdate`